<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>762</width>
    <height>522</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="1" column="1" rowspan="2">
     <widget class="QSpinBox" name="line_insert">
      <property name="buttonSymbols">
       <enum>QAbstractSpinBox::NoButtons</enum>
      </property>
      <property name="minimum">
       <number>-9999</number>
      </property>
      <property name="maximum">
       <number>9999</number>
      </property>
     </widget>
    </item>
    <item row="1" column="2">
     <widget class="QLineEdit" name="avl_tree">
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="1" column="0" rowspan="2">
     <widget class="QPushButton" name="button_insert">
      <property name="inputMethodHints">
       <set>Qt::ImhDigitsOnly</set>
      </property>
      <property name="text">
       <string>insert</string>
      </property>
     </widget>
    </item>
    <item row="0" column="0" colspan="7">
     <widget class="QGraphicsView" name="visualiztion"/>
    </item>
    <item row="2" column="2">
     <widget class="QPushButton" name="clear">
      <property name="text">
       <string>clear</string>
      </property>
     </widget>
    </item>
    <item row="1" column="3" rowspan="2">
     <widget class="QSpinBox" name="line_remove">
      <property name="buttonSymbols">
       <enum>QAbstractSpinBox::NoButtons</enum>
      </property>
      <property name="minimum">
       <number>-9999</number>
      </property>
      <property name="maximum">
       <number>9999</number>
      </property>
     </widget>
    </item>
    <item row="1" column="5" rowspan="2">
     <widget class="QPushButton" name="button_remove">
      <property name="text">
       <string>remove</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>762</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionsave"/>
   <addaction name="actionopen"/>
   <addaction name="actioninfo"/>
   <addaction name="actionabout"/>
  </widget>
  <action name="actionsave">
   <property name="text">
    <string>save</string>
   </property>
  </action>
  <action name="actionopen">
   <property name="text">
    <string>open</string>
   </property>
  </action>
  <action name="actioninfo">
   <property name="text">
    <string>info</string>
   </property>
  </action>
  <action name="actionabout">
   <property name="text">
    <string>about</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
